src/chain/tezos/TezosMessageUtil.ts

```diff
@@ -171,27 +201,91 @@ export namespace TezosMessageUtils {
     } else if (publicKey.startsWith("p2pk")) { // p256
       return "02" + base58check.decode(publicKey).slice(4).toString("hex");
     } else {
-      throw new Error("Unrecognized key type");
+      throw new Error('Unrecognized key type');
     }
   }
 
   /**
-   * Reads the key hash from the provided, bounded hex string.
-   * @param {string} hex Encoded message part.
-   * @returns {string} Key hash.
+   * Reads a key without a prefix from binary and decodes it into a Base58-check representation.
+   * 
+   * @param {Buffer | Uint8Array} b Bytes containing the key.
+   * @param hint One of 'edsk' (private key), 'edpk' (public key).
    */
-  export function readPublicKeyHash(hex: string): string {
-    if (hex.length !== 42) { throw new Error("Incorrect hex length to parse a key."); }
+  export function readKeyWithHint(b: Buffer | Uint8Array, hint: string): string {
+    const key = !(b instanceof Buffer) ? Buffer.from(b) : b;
 
-    let hint = hex.substring(0, 2);
-    if (hint === "00") { // ed25519
-      return base58check.encode(Buffer.from("0d0f25d9" + hex.substring(2), "hex"));
-    } else if (hint === "01") { // secp256k1
-      return base58check.encode(Buffer.from("03fee256" + hex.substring(2), "hex"));
-    } else if (hint === "02") { // p256
-      return base58check.encode(Buffer.from("03b28b7f" + hex.substring(2), "hex"));
+    if (hint === 'edsk') {
+      return base58check.encode(Buffer.from('2bf64e07' + key.toString('hex'), 'hex'));
+    } else if (hint === 'edpk') {
+      return readPublicKey(`00${key.toString('hex')}`);
+    } else {
+      throw new Error(`Unrecognized key hint, '${hint}'`);
+    }
+  }
+
+  export function writeKeyWithHint(b: string, hint: string): Buffer {
```

Comment:
Some of these new functions could have more specific names. It's not clear from the name what they do. Perhaps there should be a reference they are using Base58Check?